numpy/core/src/umath/scalarmath.c.src

```diff
@@ -1594,31 +1528,16 @@ static PyNumberMethods @name@_as_number = {
     (unaryfunc)@name@_negative,
     (unaryfunc)@name@_positive,                 /*nb_pos*/
     (unaryfunc)@name@_absolute,                 /*nb_abs*/
-#if defined(NPY_PY3K)
     (inquiry)@name@_bool,                       /*nb_bool*/
-#else
-    (inquiry)@name@_nonzero,                    /*nb_nonzero*/
```

Comment:
I don't believe I need to delete anything as a result of removing this use of @type@_nonzero, it seems @type@_nonzero is used in other places (e.g. [here](https://github.com/numpy/numpy/blob/master/numpy/core/src/multiarray/arraytypes.c.src#L4271))


_nonzero is used in many other places, I didn't verify each @type@_nonzero is still used (e.g. FLOAT_nonzero is but maybe cdouble_nonzero isn't)

Reply:
That is a different slot, you should fix up these lines further up in the file, that will "delete" it (not that it is ever created since we cannot compile with python 2):
```
#if defined(NPY_PY3K)
#define NONZERO_NAME(prefix) prefix##bool
#else
#define NONZERO_NAME(prefix) prefix##nonzero
#endif
```

Reply:
Thanks for finding the definition, cleaned up now.

Reply:
Does this commit remove all of those the patterns that include `NPY_PY3K` in numpy?

Reply:
Btw. I may create a merge conflict for you in descriptor.c before this goes in 10-30 minutes or so.

Reply:
Hmm, that whole macro can be simplified then of course. Dunno if you want to move that to later as well or not. The `NONZERO_NAME` macro just not necessary...

Reply:
(no problem these are easy to regenerate)

No, this doesn't clean up all uses of NPY_PY3K, there are still ~15 left that are more nuanced and I'm saving for a different PR.

Reply:
I amended ~10 more fixes into this PR (then forced pushed), if this is an anti-pattern for easy reviews let me know and I'll push each diff as a separate and squash at the end.

Reply:
Its nice to keep clean, in this instance saves me scrolling through everything again though. Since it is a large commit (and I do not know where things were changed), would be a bit nicer, but nvm.

Reply:
> Btw. I may create a merge conflict for you in descriptor.c before this goes in 10-30 minutes or so.

I just did this.